Hiring managers lessons learned (#60)
rtfd/blog: lessons-from-hiring-manager-interviews.rst

.. post:: February 1, 2019
   :tags: advertising, sustainability, hiring
   :author: David
   :location: SAN

Lessons From and For Hiring Managers
====================================

> Over the last four weeks, Eric Holscher and I did dozens of customer development interviews with engineering hiring managers.

Reviewer: ```suggestion
Over the last four weeks, the Read the Docs team did dozens of customer development interviews with engineering hiring managers.
```

We wanted to learn more about hiring processes at various companies
with the ultimate goal of building a product to help companies find developers.
We talked to people looking for talent at five person companies all the way up to the biggest names in tech.
In this post, I am going to cover some of the common things we heard from hiring managers
and **share some ways for hiring managers to improve their company's process**.

> In a week or two, I will write the second part to this blog with actionable tips

Reviewer: I've generally learned it's good to not make specific time promises. 

```suggestion
In our next post in this series, I will write the second part to this blog with actionable tips
```

Internal recruiters' time is valuable too, but we figured the hiring manager
would probably underestimate how much time the recruiter spent since they didn't directly observe it.
While answers were all over the place, the range with the most overlap was **50-100 hours of engineering time to make a hire**.
No huge revelation here, but companies make a massive investment in hiring.

From my days being an interviewer at Qualcomm and Amazon as well as a smaller startup,
I'd say fifty hours is on the low side for big company but about right for a smaller company.
When something costs 50+ engineering hours to do,
any improvement is worth a lot of time and money
and at Read the Docs we believe we can help significantly here.

Top hiring channels
-------------------

It was no big surprise to hear that **referrals were the top hiring channel**.
The second best channel was a bit of a surprise with local meetups and local communities being very high in multiple interviews
although this was less true among the largest companies we talked to.
Companies also had some success with recruiting at conferences, using various job sites, and proactive outreach.

> I'll address recruiters in a separate section.

Reviewer: Not sure this line is needed.

Hiring managers shared strategies with us that they used to increase the number of referral and community candidates they got.
Two different companies we talked to hosted multiple local meetups at their offices and one of those hiring managers said
"being plugged into the community" was their main channel for hiring.

Another hiring manager who focused a lot on developers with security experience talked about having "LinkedIn parties"
where either a hiring manager or internal recruiter sat with engineers on their team one-on-one to go through their LinkedIn network.
Even when he heard a person "is great, but they'll never change jobs", that was a good candidate to target.
Security is such a high demand field, he said, that very few great people are actively looking for a job.

.. figure:: img/2019-job-seeker-timeline.png
   :alt: Job seeker timeline
   :width: 100%

Reaching people who aren't looking is critical
----------------------------------------------

> One of our biggest takeaways at Read the Docs was that companies wanted to reach candidates before they started actively looking.

Author: ```suggestion
One of our biggest takeaways from the interviews was that companies wanted to reach candidates before they started actively looking.
```

Rather, the implication here is that as a developer becomes a mid-level developer
and then becomes a senior developer, they are more likely to rely on their network
to somewhat passively job seek rather than going through the front door
and simply applying on a company website.

Reaching people who aren't actively looking is one area where Read the Docs can really help.
Companies want to reach them when they are just thinking about a new job or perhaps passively seeking.
Developers who come to our site are actively building software, not job-seeking.
Many of them are content at their current position.
For those that aren't, this is a perfect place to showcase great companies
and help Read the Docs' visitors to find their next job.

.. _Joel Spolsky had it right: https://www.inc.com/magazine/20070501/column-guest.html

Recruiters can be worth it
--------------------------

In each interview, we asked whether the company used recruiters or recruiting related services (eg. Hired, Triplebyte).
Universally, hiring managers brought up how expensive recruiters are,
but most of the people we talked to who had actually used recruiters in the past had a positive impression overall.

> Hiring takes a ton of time and typically takes it from senior folks at the company.

Reviewer: ```suggestion
Hiring takes a ton of time, typically from senior folks at the company.
```

Even though the price is high, recruiters who help speed that process along are worth it.

I think the quote that really summed it up was from a VP of engineering at a startup that just raised a Series B.
He's trying to grow his team by 10 developers this year (about a 50% increase).
His experience hiring his first couple developers through a 2-person boutique recruiting firm was great and
he would use them for hiring everyone "if hiring 10 people didn't mean $100k+" in fees alone.

Full disclosure: Triplebyte is an advertiser on Read the Docs.

Filtering resumes isn't a real problem
--------------------------------------

Every single hiring manager we talked to complained about the time they spent filtering resumes from applicants.
Some choice quotes I heard included "95% of applicants weren't a fit" even after only reviewing the resume
and a hiring manager who posted to a very popular remote-only job board told me he got "literally one thousand applications".
While I believe this is a legitimate complaint,
I think `Aline Lerner said it best`_ when she said "Engineering hiring isn't a filtering problem. It's a sourcing problem."

> Filtering is just a time-consuming task that hiring managers directly deal with.

Reviewer: This feels a bit dismissive and negative, especially the heading. Perhaps "Sourcing people is the real problem" is a better angle.

The real problem here is that **the number of applicants to a job post is a vanity metric**.
What hiring managers are looking for is qualified applicants, which is harder to gauge without some additional work.
One of the problems here is that a lot of jobs sites make it really easy for candidates to apply to lots of jobs.
This is understandable from their perspective as more applicants makes it look like their platform is more valuable.

> I'm not trying to talk trash on these platforms as there's no reason for them to erect artificial barriers to the application funnel,

Reviewer: This feels a bit weird and defensive. Probably a better way to say this.

but hiring managers told us they want candidates who want to work for their company
not somebody who is applying to every single job out there. More on that in a future post.

Instead of building product here, I think hiring managers just need a few tweaks
to get more of a signal of candidate quality and intent in their applications.
A few of the hiring managers we talked to described how they did this.

> Multiple hiring managers mentioned how they required a cover letter that answered specific questions ("why us" was the most popular).

Reviewer: Might be good to put these in a list to make it easier to skim. Otherwise it will likely get caught on a skim.

Another hiring manager described how his hiring system just asked a few short questions of candidates
-- for example, "what is the software project you're most proud of" -- when they applied.
When a candidate didn't do a cover letter or didn't answer the questions, they weren't considered.
I think the latter solution is best as it solves the one-click apply problem without being too much of a barrier.

.. _Aline Lerner said it best: http://blog.alinelerner.com/building-a-product-in-the-technical-recruiting-space-read-this-first/

> TLDR

Reviewer: I've a bit less excited about TLDR -- perhaps "In summary"

----

Every company we talked to said that hiring talent and especially senior developers was hard, expensive, and time-consuming.
At the same time, many of the hiring managers we talked to had tactics they used to improve the process and you can use these strategies too.

* **Have a strategy for getting more referrals** whether that is actively asking employees or having an overly generous referral bonus.
  Referrals are probably your top source of candidates so figuring out how to get more will save you time and money in the end.
  The same goes for any of your other best recruiting channels.
* **Reach passive candidates**. There are a lot more passive candidates than active ones so have a plan to reach them.
  Great developers (especially senior ones) probably aren't actively looking for a new job and it's a job-seekers market.
  Even if somebody isn't considering a new job at all, the next time they do, you want them to think of your company.
* **Get more signal in your application process** by adding a step such as a cover letter, quiz,

>   or simply add a couple questions folks have to answer.

Reviewer: I wonder if we should put this at the top (where a tl;dr would normally go) as an intro, in order to get people to read further. I feel like this post is probably too long for most people to read, so having more of a hook would be good.
